### PR TITLE
Fun token example

### PR DESCRIPTION
**Pull request by evgenykuzyakov**
I need a test that has different context within a single test, so I can test locks and allowance properly. @nearmax  

Also looks like I need to update comments.

### CODE REVIEW

File: examples/fun-token/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,502 @@
...
+        if account.balance < lock_amount {
+            env::panic(b"Not enough unlocked balance");
+        }
+        account.balance -= lock_amount;
+
+        // If locking by escrow, need to check and update the allowance.
+        if escrow_account_id != owner_id {
+            let allowance = account.get_allowance(&escrow_account_id);
+            if allowance < lock_amount {
+                env::panic(b"Not enough allowance");
```
Comment (ailisp) [inline examples/fun-token/src/lib.rs]: i have a question here, if paniced here the before `account.balance -= lock_amount` won't commit?

File: examples/fun-token/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+    }
+    fn bob() -> AccountId {
+        "bob.near".to_string()
+    }
+    fn carol() -> AccountId {
+        "carol.near".to_string()
+    }
+
+    fn get_context(predecessor_account_id: AccountId) -> VMContext {
+        VMContext {
```
Comment (ailisp) [inline examples/fun-token/src/lib.rs]: these fixtures seems appear a lot of place, would be helpful to provide as a helper?

File: examples/fun-token/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,502 @@
...
+        testing_env!(get_context(bob()), Config::default());
+        contract.lock(carol(), lock_amount);
+        assert_eq!(contract.get_allowance(carol(), bob()), allowance - lock_amount);
+        assert_eq!(contract.get_unlocked_balance(carol()), total_supply - lock_amount);
+        assert_eq!(contract.get_total_balance(carol()), total_supply);
+        // Acting as alice now
+        testing_env!(get_context(alice()), Config::default());
+        contract.lock(carol(), lock_amount);
+    }
+}
```
Comment (ailisp) [inline examples/fun-token/src/lib.rs]: :+1 a lot of interesting cases

File: examples/fun-token/src/lib.rs (hunk 1 above)
Comment (evgenykuzyakov) [reply examples/fun-token/src/lib.rs]: Yes

File: examples/fun-token/src/lib.rs (hunk 2 above)
Comment (evgenykuzyakov) [reply examples/fun-token/src/lib.rs]: We can provide some default implementation and just modify details.

**Pull request merged**